internal/crypto/ptls/fips_strict.go

```diff
@@ -77,3 +78,11 @@ func DefaultLDAP(rootCAs *x509.CertPool) *tls.Config {
 func secureServing(opts *options.SecureServingOptionsWithLoopback) {
 	defaultServing(opts)
 }
+
+// SetAllowedCiphersForTLSOneDotTwo does not allow users to specify allowedCipherNames in FIPS mode
+func SetAllowedCiphersForTLSOneDotTwo(allowedCipherNames []string) error {
+	if len(allowedCipherNames) > 0 {
+		return fmt.Errorf("FIPS mode does not accept allowedCipherNames")
```

Comment:
I think we should make this feature work equally well for FIPS builds too. There's no extra difficulty and it would make the feature available for those downstream teams which choose to compile in FIPS mode.

Reply:
Updated accordingly. FIPS is not as well tested either through unit or integration tests, so I'm less confident that it works all together as expected.